app/models/service_orchestration.rb

```diff
@@ -27,14 +27,14 @@ def orchestration_stack_status
   end
 
   def deploy_orchestration_stack
-    deploy_stack_options = stack_options
     job_options = {
-      :create_options            => deploy_stack_options,
+      :create_options            => stack_options,
       :orchestration_manager_id  => orchestration_manager.id,
       :orchestration_template_id => orchestration_template.id,
       :stack_name                => stack_name,
       :zone                      => my_zone
     }
+    job_options[:execution_ttl] = stack_options[:execution_ttl] || options[automate_timeout_key('provision')]
```

Comment:
Would recommend moving the to calls into a private method to dry up the local and add tests.